Framework/API/inc/MantidAPI/SpectrumDetectorMapping.h

```diff
@@ -53,7 +47,7 @@ class MANTID_API_DLL SpectrumDetectorMapping {
   using sdmap = std::unordered_map<specnum_t, std::set<detid_t>>;
 
 public:
-  explicit SpectrumDetectorMapping(const MatrixWorkspace *const workspace,
+  explicit SpectrumDetectorMapping(MatrixWorkspace_const_sptr &workspace,
```

Comment:
While at it, maybe you could mark `useSpecNoIndex` as `const` as well?

Reply:
I think you misunderstood me. I'm just suggesting changing `bool useSpecNoIndex` to `const bool useSpecNoIndex`. It has no effect on default values or initialization whatsoever.

Reply:
You're right.